src/renderer/views/GridFeed.js

```diff
@@ -4,24 +4,68 @@ import VideoWidget from "../components/video/VideoWidget";
 class GridFeed extends React.Component {
     constructor() {
         super();
-        this.state = { data: [] };
+        this.state = { data: [], lazyLoading: false };
+        this.handleScroll = this.handleScroll.bind(this);
     }
 
     componentDidMount() {
+        window.addEventListener("scroll", this.handleScroll);
         fetch(`https://3speak.online/apiv2/feeds/${this.props.type}`)
             .then(res => res.json())
             .then(json => {
                 this.setState({data: json})
             })
     }
 
+    componentWillUnmount() {
+        window.removeEventListener("scroll", this.handleScroll);
+    }
+
+    handleScroll() {
+        if (!this.state.lazyLoading) {
```

Comment:
Could you modify to use `this.props` instead of `this.state`? So parent components can pass options directly when creating an instance.